clang/lib/Sema/SemaDecl.cpp

```diff
@@ -3217,6 +3217,22 @@ static void adjustDeclContextForDeclaratorDecl(DeclaratorDecl *NewD,
     FixSemaDC(VD->getDescribedVarTemplate());
 }
 
+template <typename AttributeName>
+static void checkDimensionsAndSetDiagnostics(Sema &S, FunctionDecl *New,
+                                             FunctionDecl *Old) {
+  auto *NewDeclAttr = New->getAttr<AttributeName>();
```

Comment:
Can we be more specific and don't use `auto` here?

Reply:
Ok